src/category_theory/endofunctor/algebra.lean

```diff
@@ -0,0 +1,200 @@
+/-
+Copyright (c) 2022 Joseph Hua. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison, Bhavik Mehta, Johan Commelin, Reid Barton, Rob Lewis, Joseph Hua
+-/
+import category_theory.functor_category
+import category_theory.concrete_category.reflects_isomorphisms
+import category_theory.limits.final
+
+/-!
+# Algebras of endofunctors
+
+This file defines algebras of an endofunctor,
+and provides the category instance for them.
+This extends to Eilenberg-Moore (co)algebras for a (co)monad.
+
+It also defines the forgetful functor from the category of algebras.
+It is shown that the structure map of the initial algebra of an endofunctor
+is an isomorphism.
+
+-/
+
+universes v u
+
+namespace category_theory
+namespace endofunctor
+
+variables {C : Type u} [category.{v} C]
+
+/-- An algebra of an endofunctor; `str` stands for "structure morphism" -/
+structure algebra (F : C ⥤ C) :=
+(A : C)
+(str : F.obj A ⟶ A)
+
+instance [inhabited C] : inhabited (algebra (𝟭 C)) := ⟨⟨ default , 𝟙 _ ⟩⟩
+
+namespace algebra
+
+variables {F : C ⥤ C} (A : algebra F) {A₀ A₁ A₂ : algebra F}
+
+/-
+        str
+   F A₀ -----> A₀
+    |          |
+F f |          | f
+    V          V
+   F A₁ -----> A₁
+        str
+-/
```

Comment:
I think that commutative diagrams like this don't show up in the API docs unless you quote them. Is my suggestion correct?
```suggestion
/-
```
        str
   F A₀ -----> A₀
    |          |
F f |          | f
    V          V
   F A₁ -----> A₁
        str
```        
-/
```

Reply:
darn, `suggestion` also uses triple backticks. You know what I mean :-) How does one make a suggestion containing a line with triple back ticks in?

Reply:
@kbuzzard I think you need to use quadruple backticks for the outer set.